### PR TITLE
Update for latest wkt

### PR DESCRIPTION
**Pull request by kylebarron**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -56,7 +61,7 @@ serde_json = "1.0.104"
 sqlx = { version = "0.7", default-features = false }
 thiserror = "1.0"
 tokio = { version = "1.30.0", default-features = false }
-wkt = "0.10.3"
+wkt = { git = "https://github.com/georust/wkt", rev = "a26848c97686245548a43375133f41e77322247d" }
```
Comment (nyurik) [inline Cargo.toml:64]: you can just add this to the `patch.crates-io` section below. Not a biggie for a WIP of course

**Review by nyurik — COMMENTED** [review]
seems reasonable

File: geozero/src/wkt/wkt_reader.rs [hunk 2]
```diff
@@ -109,7 +108,7 @@ pub(crate) fn process_wkt_geom_n<P: GeomProcessor>(
                 processor.empty_point(idx)
             }
         }
-        Geometry::MultiPoint(g) => {
+        wkt::Wkt::MultiPoint(g) => {
```
[the remark this answers is not in the document]
Comment (michaelkirk) [reply geozero/src/wkt/wkt_reader.rs:111]: This doesn't strike me as weird. It's an enum variant with an explicit module.

The explicit module is usually overly verbose, so you can get rid of it, but in this case there are already other entities in the local module scope called `Wkt`. I think this is good as is.

**Review by michaelkirk — APPROVED** [review]

File: geozero/src/wkt/wkt_reader.rs [hunk 3]
```diff
@@ -85,22 +84,22 @@ pub fn read_wkt<R: Read, P: GeomProcessor>(reader: &mut R, processor: &mut P) ->
     let mut wkt_string = String::new();
     reader.read_to_string(&mut wkt_string)?;
     let wkt = wkt::Wkt::from_str(&wkt_string).map_err(|e| GeozeroError::Geometry(e.to_string()))?;
-    process_wkt_geom(&wkt.item, processor)
+    process_wkt_geom(&wkt, processor)
 }
 
 /// Process WKT geometry
-fn process_wkt_geom<P: GeomProcessor>(geometry: &Geometry<f64>, processor: &mut P) -> Result<()> {
+fn process_wkt_geom<P: GeomProcessor>(geometry: &wkt::Wkt<f64>, processor: &mut P) -> Result<()> {
```
[the remark this answers is not in the document]
Comment (michaelkirk) [reply geozero/src/wkt/wkt_reader.rs:91]: there are already other entities in the local module scope called Wkt. You could use an alias, but it's unlikely to be any more concise than wkt::Wkt.

File: geozero/src/wkt/wkt_reader.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (nyurik) [reply geozero/src/wkt/wkt_reader.rs:111]: @michaelkirk to clarify - wkt:: thing is nice to have but totally optional - it was more of a question really.  I would still prefer not to reformat Cargo.toml though.

File: Cargo.toml [hunk 4]
```diff
@@ -27,7 +30,9 @@ clap = { version = "4.3", features = ["derive"] }
 criterion = "0.5.1"
 csv = "1.2.2"
 dbase = "0.4"
-diesel = { version = "2.1.0", default-features = false, features = ["postgres"] }
+diesel = { version = "2.1.0", default-features = false, features = [
```
[the remark this answers is not in the document]
Comment (kylebarron) [reply Cargo.toml:33]: Sorry this was auto-formatted when I saved the file

File: geozero/src/wkt/wkt_reader.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (michaelkirk) [reply geozero/src/wkt/wkt_reader.rs:111]: Sorry, forgot about this. It looks like @kylebarron reverted the automated Cargo.toml format changes, so I think everything is LGTM. Can you approve @nyurik so we can merge?

**Review by michaelkirk — APPROVED** [review]

**Review by nyurik — APPROVED** [review]
thx!!

**Comment by nyurik** [comment]
When merged, will all the minor back and forth commits go into the main tree?

**Comment by kylebarron** [comment]
I usually always squash commits when merging myself

**Comment by michaelkirk** [comment]
I switched the merge queue to use squash commits just now.

For the record, as a staunch "rebase and merge-commit"-er, I find squashing barbaric, but barbarians have proven themselves very effective over the course of history. 😉 

Another maintainer can LMK if they care. I'm open to undoing that change, but figured I'd rather be transparent about the change and ask for foregiveness than have a drawn out debate beforehand.

**Comment by michaelkirk** [comment]
Looks like new clippy is hitting new lint. I'll follow up with a fix.

**Comment by nyurik** [comment]
I wonder if clippy should be a non voting part of ci? Ie ok to merge, but show up as an error. 

-- your friendly barbarian

**Comment by michaelkirk** [comment]
> I wonder if clippy should be a non voting part of ci? Ie ok to merge, but show up as an error.

I'm slightly in favor of that change, though don't feel strongly. 

Obviously this opens us up the risk of ignoring important clippy lints, but on balance, I'd support that change. 

It's disruptive that I frequently lose part of my day trying to move something along. Often it's just 5 minutes, but it's not uncommon for it to be 20 or more. It's especially annoying because IMO 80% of the time the clippy changes are neutral, 10% of the time they are dumb, and 10% of the time they are super helpful. 

To be clear, I'm pro-clippy. It's saved my bacon many times. I just don't like something with such a low signal to noise ratio scheduling my day for me.

**Pull request merged**